webapp/lib/app/configurator/standard_messages/controller_messages_helper.dart

```diff
@@ -9,81 +22,184 @@ class StandardMessagesManager {
+    localCategories.putIfAbsent(standardMessage.categoryId, () => new MessageCategory(standardMessage.categoryId, standardMessage.category, standardMessage.categoryIndex));
+    localCategories[standardMessage.categoryId].groups.putIfAbsent(standardMessage.groupId, () => MessageGroup(standardMessage.groupId, standardMessage.groupName, standardMessage.groupIndexInCategory));
+    localCategories[standardMessage.categoryId].groups[standardMessage.groupId].messages[standardMessage.suggestedReplyId] = standardMessage;
     return standardMessage;
   }
 
-  List<model.SuggestedReply> addStandardMessages(List<model.SuggestedReply> standardMessagesToAdd) {
+  model.SuggestedReply onAddStandardMessageFromStorage(model.SuggestedReply standardMessage) {
+    storageCategories.putIfAbsent(standardMessage.categoryId, () => MessageCategory(standardMessage.categoryId, standardMessage.categoryName, standardMessage.categoryIndex));
+    storageCategories[standardMessage.categoryId].groups.putIfAbsent(standardMessage.groupId, () => MessageGroup(standardMessage.groupId, standardMessage.groupName, standardMessage.groupIndexInCategory));
+    storageCategories[standardMessage.categoryId].groups[standardMessage.groupId].messages.putIfAbsent(standardMessage.suggestedReplyId, () => standardMessage.clone());
+    return standardMessage;
+  }
+
+  List<model.SuggestedReply> onAddStandardMessagesFromStorage(List<model.SuggestedReply> standardMessagesToAdd) {
     List<model.SuggestedReply> added = [];
     for (var standardMessage in standardMessagesToAdd) {
-      var result = addStandardMessage(standardMessage);
+      var result = addStandardMessageInLocal(standardMessage);
+      onAddStandardMessageFromStorage(standardMessage);
       if (result != null) added.add(result);
     }
     return added;
   }
 
-  model.SuggestedReply updateStandardMessage(model.SuggestedReply standardMessage) {
-    var removed = removeStandardMessage(standardMessage);
-    var updated = addStandardMessage(standardMessage);
+  model.SuggestedReply updateStandardMessageInLocal(model.SuggestedReply standardMessage) {
+    var removed = removeStandardMessageInLocal(standardMessage);
+    var updated = addStandardMessageInLocal(standardMessage);
     if (removed == null || updated == null) {
       throw "Standard message consistency error: The two-step update process has found an inconsistency for updating tag ${standardMessage.suggestedReplyId}";
     }
     return updated;
   }
 
-  List<model.SuggestedReply> updateStandardMessages(List<model.SuggestedReply> standardMessages) {
-    List<model.SuggestedReply> updated = [];
-    for (var standardMessage in standardMessages) {
-      var result = updateStandardMessage(standardMessage);
-      if (result != null) updated.add(result);
+  model.SuggestedReply onUpdateStandardMessageFromStorage(model.SuggestedReply standardMessage) {
```

Comment:
nit - let's rename this slightly and make it a private method: `_updateStandardMessageInStorage`